libs/config/node.go

```diff
@@ -0,0 +1,106 @@
+package config
+
+import "time"
+
+type Value struct {
+	v any
+	l Location
+
+	// Whether or not this value is an anchor.
+	// If this node doesn't map to a type, we don't need to warn about it.
+	anchor bool
+}
+
+var NilValue = Value{
+	v: nil,
+	l: Location{},
+}
+
+func New(v any, loc Location) Value {
+	return Value{
+		v: v,
+		l: loc,
+	}
+}
+
+func (v Value) AsMap() (map[string]Value, bool) {
+	m, ok := v.v.(map[string]Value)
+	return m, ok
+}
+
+func (v Value) Location() Location {
+	return v.l
+}
+
+func (v Value) AsAny() any {
+	switch vv := v.v.(type) {
+	case map[string]Value:
+		m := make(map[string]any)
+		for k, v := range vv {
+			m[k] = v.AsAny()
+		}
+		return m
+	case []Value:
+		a := make([]any, len(vv))
+		for i, v := range vv {
+			a[i] = v.AsAny()
+		}
+		return a
+	case string:
+		return vv
```

Comment:
should it be `v.v` instead? `vv` seems to be the type of the value

Reply:
It's valid, see https://go.dev/tour/methods/16. It's a special case of switch that allows for type matching.